### PR TITLE
Room navigation

### LINKED ISSUE #20: Add ability to move between rooms
### Expected behavior
Once the character enters a room's doorway they should be moved to the connected room.

### Actual behavior
Player can't move to another room

### PR DESCRIPTION
**Pull request by munkeyxis**
Closes #20 

@whoa-algebraic/jake

### CODE REVIEW

**Comment by munkeyxis** [comment]
Important to note if you test it in Unity. When switching rooms, the camera does not snap to the player character's new location. This is an issue that'll be addressed soon and is detailed here #19 

Edit: you can see the character in the new room by continueing to move left until the character leaves the "buffer area" which will cause the camera to snap to the character again.

File: Assets/Scripts/RoomNavigationManager.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        Destroy(activeRoomPrefab);
+
+        // spawn connectedRoom and save reference
+        activeRoomPrefab = Instantiate(GetRoomPrefabById(connectedRoomId));
+
+        // place player near connected door
+        // find position of connected door
+        Vector3 connectedDoorPosition = GetDoorPosition(connectedDoorId);
+
+        // determin if player should be on the left or right side of the door's location
```
Comment (adam-balazs) [inline Assets/Scripts/RoomNavigationManager.cs:27]: typo: determine :P

File: Assets/Scripts/RoomNavigationManager.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    // assumes that a valid ID is provided, if not bad things will happen
+    // TODO how can this be done safer, or throw an error if no room is found with the given ID?
+    GameObject GetRoomPrefabById(int id) {
+        GameObject roomPrefab = new GameObject();
+
+        // loop through all the room prefabs
+        foreach(GameObject room in RoomPrefabs) {
+            if(room.GetComponent<RoomDetails>().Id == id) {
+                roomPrefab = room;
```
Comment (adam-balazs) [inline Assets/Scripts/RoomNavigationManager.cs:42]: you can return here. If we get more rooms, this foreach can run for a long time. Might be a good idea to have an IdToRoom map for better performance

File: Assets/Scripts/RoomNavigationManager.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    // assumes that a valid ID is provided, if not bad things will happen
+    // TODO how can this be done safer, or throw an error if no room is found with the given ID?
+    // TODO perhaps change roomDetail's Door array into an array of DoorDetails rather than GameObjects
+    Vector3 GetDoorPosition(int doorId) {
+        Vector3 doorPosition = Vector3.zero;
+
+        foreach(GameObject door in activeRoomPrefab.GetComponent<RoomDetails>().Doors) {
+            if(door.GetComponent<DoorDetails>().Id == doorId) {
+                doorPosition = door.transform.position;
```
Comment (adam-balazs) [inline Assets/Scripts/RoomNavigationManager.cs:57]: same applies as before. Might worth to introduce an IdToDoor map

**Pull request merged**